src/main/webapp/script.js

```diff
@@ -100,21 +100,84 @@ window.addEventListener('load', () => {
 
   getOptions();
 
-  function initCalendar() {
-    new tui.Calendar('#calendar', {
-      defaultView: 'week',
-      useCreationPopup: true,
-      useDetailPopup: true,
-      disableDblClick: true,
-      disableClick: true,
-      isReadOnly: true,
-      scheduleView: ['time'],
-      taskView: false,
-      week: {
-        workweek: true,
-      },
+  if (document.head.id == 'scheduler') {
```

Comment:
I feel like this is kinda strange. I think it would be better to have separate js files for each html page and extract the reused functions into its own js file. Then you could just import that file at the top of this file and use the functions. I've just never seen js that depends on the head's id, but let me know if doing it this way is normal.

Reply:
the problem that I saw with that is I wouldn't be able to make those functions and variables I want to import anonymous, which could potentially cause the namespace collisions that William mentioned.

Reply:
Great points from both of you. Some thoughts:

1. Instead of document head id (i've also never seen this before), we could:
  a. do this only when the calendar element (`#calendar`) exists [I would lean towards this if we're going with the 1-javascript-file approach]
  b. inspect the url of the page
2. We chatted a bit about the new file option during our sync today. We can reduce namespace collisions by "owning" a top-level namespace (`CollegePlanner.*`) that is very unlikely to collide with anything. So you could create a `CollegePlanner.CourseSelector` component for example. This is a bit more sustinable since we can just create more files as different use cases come up.